backend/subroker/views.py

```diff
@@ -98,459 +140,416 @@ def login(request):
+            is_public=group_is_public,
+            password=group_password,
+            membership=group_ott_plan,
+            payday=group_payday,
+            account_bank=group_account_bank,
+            account_number=group_account_number,
+            account_name=group_account_name,
+            leader=request.user,
+        )
+        group.save()
+
+        group.members.add(request.user.id)
+        group.save()
+
+        response_dict = {
+            'id': group.id,
+            'name': group.name,
+            'leader': group.leader.id
+        }
+        return JsonResponse(response_dict, status=201)
+
+@require_http_methods(["GET", "PUT", "DELETE"])
+@login_required
 def group_detail(request, group_id):
+    """
+    /api/group/<int:group_id>/
+
+    GET
+        Get group detail information
+
+    PUT
+        Edit group detail information
+
+    DELETE
+        delete group
+    """
     if request.method == 'GET':
-        if request.user.is_authenticated:
-            try:
-                group = Group.objects.get(id=group_id)
-            # ERR 404 : Group Doesn't Exist
-            except (Group.DoesNotExist) as _:
-                return HttpResponse(status=404)
-            leader = User.objects.filter(id=group.leader.id).values()[0]
-            members = [
-                {
-                    "id": member.id,
-                    "username": member.username,
-                }
-                for member in group.members.all()]
-            response_dict = {
-                "id": group.id,
-                "name": group.name,
-                "platform": group.membership.ott,
-                "membership": group.membership.membership,
-                "cost": group.membership.cost,
-                "maxPeople": group.membership.max_people,
-                "currentPeople": group.current_people,
-                "members": members,
-                "accountBank": group.account_bank,
-                "accountNumber": group.account_number,
-                "accountName": group.account_name,
-                "description": group.description,
-                "payday": group.payday,
-                "leader": leader,
+        try:
+            group = Group.objects.get(id=group_id)
+        # ERR 404 : Group Doesn't Exist
+        except (Group.DoesNotExist) as _:
+            return HttpResponse(status=404)
+
+        leader = User.objects.get(id=group.leader.id)
+        members = [
+            {
+                "id": member.id,
+                "username": member.username,
             }
-            return JsonResponse(response_dict, safe=False, status=200)
-        # ERR 401 : METHOD NOT ALLOWED
-        else:
-            return HttpResponse(status=401)
+            for member in group.members.all()]
+
+        response_dict = {
+            "id": group.id,
+            "name": group.name,
+            "platform": group.membership.ott,
+            "membership": group.membership.membership,
```

Comment:
this would be conflict, since we need isPublic, password info also. just to remind me